### ISSUE TITLE
Stop calling it Open Source if you only provide Commons Clause license

### ISSUE THREAD

**Issue by nealmcb** [ComodoSecurity/openedr#4]
Your README and the press are calling this _Open Source_ software.
But your [License.md](https://github.com/ComodoSecurity/openedr/blob/main/LICENSE.md) is the [Commons Clause](https://commonsclause.com/) which explicitly says it is not Open Source:

> Is this “Open Source”?
> No.
> “Open source”, has a specific definition that was written years ago and is stewarded by the Open Source Initiative, which approves Open Source licenses. Applying the Commons Clause to an open source project will mean the source code is available, and meets many of the elements of the Open Source Definition, such as free access to source code, freedom to modify, and freedom to re-distribute, but not all of them.

The difference is of course major for anyone who wants to join you in providing OpenEDR as a product.

Please fix your README.

**Comment by ComodoMelih** [comment]
hi Neal, 
We will edit the readme to read:
"Open Source with Common Clause" 
thanks for the heads up.

**Comment by nealmcb** [comment]
You can call it "_source-available_" or "_disclosed source_" or something else.

But it is **not** "_open source_", and the license you use clearly states that, as I quoted above.

You are retaining the right to sue people e.g. for just selling _consulting_ services on how to deploy or use OpenEDR:

> ... the License does not grant to you, the right to Sell the Software.
> For purposes of the foregoing, “Sell” means practicing any or all of the rights granted to you under the License to provide to third parties, for a fee or other consideration (including without limitation fees for hosting or consulting/ support services related to the Software), a product or service whose value derives, entirely or substantially, from the functionality of the Software.

You can't add the Commons Clause to a license and still claim that it is open source. It would be like claiming that something is "_[USDA Certified Organic](https://en.wikipedia.org/wiki/Organic_certification)_ with added pesticides"

**Comment by ComodoMelih** [comment]
> You are retaining the right to sue people e.g. for just selling _consulting_ services on how to deploy or use OpenEDR:

My understanding was if someone took the code, created a derivative software and then charged for that derivative software was the issue. Using OpenEDR to offer a service wasn't caught by this license. That is my understanding and will double check it with legal people.

**Comment by nealmcb** [comment]
Until these issues are properly resolved, please re-open this issue.

## Right to sell Open Source is guaranteed
The right to sell e.g. Apache or Linux, or a derivative of them, or consulting around them, is a major reason why people seek out open source software and shun software with entanglements like the Commons Clause. The right to do that with _any_ open source code is explicitly guaranteed by the [The Open Source Definition \(Annotated\) \| Open Source Initiative](https://opensource.org/osd-annotated):

 > 1. Free Redistribution
The license shall not restrict any party from selling or giving away the software

So for that reason alone, **you must refrain from claiming that OpenEDR is Open Source**.

## Right to offer OpenEDR via SaaS is not granted given the Commons Clause

Despite the reference to allowing others to offer SaaS in your quote above, you also must be clear that no one else is allowed to sell OpenEDR as a service in a SaaS offering. As the FAQ also clarifies:

> Can you change the name of the library and change some function names and sell the library or offer it as SaaS?
No.

## Right to consult on OpenEDR seems explicitly prohibited by Commons Clause

Re offering **consulting services**, the FAQ is not a legal interpretation, as even author of the clause, Heather Meeker, says at the "clarifying discussion" link:

> Of course, I'm not your lawyer, so technically I can't give you advice. The clause means what it means, and although I led the drafting of the clause, that doesn't mean I have authority to interpret documents. That's not how the law works -- in the end, only a court has that power, no matter who wrote the document.

Furthermore, that "_clarifying discussion_" is a GitHub issue entitled [Preventing specialized consultants?](https://github.com/fossas/commons-clause/issues/4) which is well worth reading very carefully.  Despite the fact that the issue was closed, it leaves me, the author of the issue (saibotsivad), and many others very dubious about why the text of the license wouldn't also apply to any consultant.

How else could the inclusion of the phrase _"consulting/support services "_ in the text of the license be interpreted?

Given that legal uncertainty, if you don't want to cut your product off from the benefits of having others offer consulting around it,
you should also refrain from using the Commons Clause itself.

## Importance of Community to open source software projects, vs Business Models

Open Source is about supporting the **community**.  Commons Clause is about giving one particular organization (the copyright holder, Comodo here) broad rights that no one else has.

See more clarity on the focus Open Source has on _empowering communities_, vs various non-open-source attempts to support various _business models_ at [We need Sustainable Free and Open Source Communities](https://medium.com/sustainable-free-and-open-source-communities/we-need-sustainable-free-and-open-source-communities-edf92723d619)  by Adam Jacob \| Sustainable Free and Open Source Communities \| Medium

**Comment by ComodoMelih** [comment]
> Right to offer OpenEDR via SaaS is not granted given the Commons Clause
If you do nothing but take the code and add no value and redistribute it as SaaS, then I agree with you. However if you offer it within an offering etc its ok (please read below). So any SOC/MSSP can use OpenEDR for free as part of their service for example, or any enterprise can use OpenEDR in their environment for free. **That is our intention.** 
![image](https://user-images.githubusercontent.com/74210954/99159621-ef3dcd00-26ab-11eb-9a79-cdaa8334b1dc.png)

Although I am not a lawyer but to me the question of : "consulting" on "OpenEDR" is very clear.
![image](https://user-images.githubusercontent.com/74210954/99159701-f0bbc500-26ac-11eb-8966-df6b2d6f367c.png)
Yes you can consult on OpenEDR and **That is our intention.**

So I answered both question and also "confirmed publicly" that it was the intention of the documents. So this public document, if needed, can be used in the court of law.

When it comes to about "Importance of Community to open source software projects, vs Business Models"
I personally believe Commons Clause forces the hand of any company who want to benefit from "open code" to contribute back to the community.  I believe more money/investment going into "Open Code" is a good thing and I believe Commons Clause helps achieve that. The difference is individuals contributing vs corporations contributing. With the right model we can get the corporations contributing to opening the code and not just individuals! There are too many corporations out there that benefit from Open Code without contributing back! Commons Clause can help reverse that! (these are my subjective views).

**Comment by ComodoMelih** [comment]
Is the phrase "Open Source" owned by any trademark owner?

**Comment by nealmcb** [comment]
Thank you for your quick and helpful responses. I am glad to see your movement towards more openness, especially in security-relevant software like yours.

I'm an open source and security expert, not a trademark expert. Regardless of the legalities of usage of the term, it has long been clearly understood to mean software that meets the Open Source Initiative definition, and that definition is itself referenced by the Commons Clause FAQ. More particularly, I'm confident that if your intent is to engage the software community, you should abide by the clear statements in the FAQ and by the author of the Commons Clause that you should not describe your software as "open source" while adding the Commons Clause.

Given the discussion, it is still my opinion that a different "rider" than the Commons Clause might make sense if your goal is specificially to prohibit selling OpenEDR. The clause text and FAQ also seems quite clear that a SOC or MSSP offering a SaaS service which relies on unmodified software would be "substantially" relying on the value of OpenEDR, thus putting them in legal limbo at best. No offense, but I'd hate to be pulled in front of a judge by some company that might buy out Comodo some day, who has a different sensibility about this, and having defend myself by pointing to unsigned, editable comments in a GitHub issue, even though they are written by someone who appears to be the CEO.

Finally, I see no indication of which underlying open source license you are modifying by the Commons Clause, since the latter is the only licensing information noted in the License.md file.

**Comment by ComodoMelih** [comment]
Thanks for your valuable input Neal, these kind of open discussions are what helps us all improve. 

Open sourcing/Open source vs Open Source Initiative: 
My understanding is Open Source Initiative is a TM ( https://opensource.org/trademark-guidelines ) and can't use the exact phrase Open Source Initiative unless you comply with its requirements.
it is my understanding that "verbs" like "Open sourcing, Open sourced" etc are not covered by that Trademark and can be used.

Is it your understanding that one can't use the verb "Open sourced" because of the Trademark? If so can you point me to a trademark on that please?

I validated with our legal guy that with our Commons Clause a SOC or MSSP can offer SaaS without modifying software, because they are adding their services. Legally they are not seeing any issues with that and they suggested if you want a clarification, I can send you what I stated publicly that its ok for a SOC or MSSP to use it as is as part of their services without modifying the code on an email (as the CEO of the company confirming that). If you let me know your email address happy to send that email.

**Comment by nealmcb** [comment]
I think this is about clear documentation, avoiding confusion, and putting user's minds at ease. Several issues remain in play:

* The Commons Clause is not a license, and only restricts usage further. Until you actually pick a license which gives people some rights to the code, and document it in the LICENSE.md, the code remains completely proprietary. See also https://twitter.com/spendergrsec/status/1326951779628687362

* Again, why would you use a legal document like the Commons Clause, and then do the opposite of what it says? It says not to describe OpenEDR as "open source", so doing so is at best misleading. It isn't a matter of trademarks, it is a matter of decades of consensus on the meaning in the community, clear communication, and making sure your users know what they can and can't do. Misusing the term just gives you a black eye at exactly the time when you could be winning praise for opening things up. You'll just waste your time, our time, and potential legal hassles down the road when people rely on the consensus meaning of "open source" and do things that you don't want to allow. Surely we all want to spend our resources writing code and defending enterprise security, rather than hiring lawyers.

* Again, if you want to encourage SOC/MSSP usage, please note the many voices of confusion on the FAQ that I linked above, and try to find a way to more definitively encourage what you want, and discourage what you prohibit, without scaring away your users and _their_ lawyers. Given that there are other fully open source EDR solutions out there, I'd suggest you would want to find a way to be maximally welcoming and reassuring in order to draw users to your solution.

**Comment by ComodoMelih** [comment]
> I think this is about clear documentation, avoiding confusion, and putting user's minds at ease. Several issues remain in play:

I agree with that statement. 

Let me talk to the team again to make sure all these issues are addressed. 
thanks for raising them Neal!

**Comment by ComodoMelih** [comment]
https://github.com/ComodoSecurity/openedr/blob/main/LICENSE.md
Until we discuss it with the team...I updated the above link. (Underlying agreement is GPL3...for whatever reason it wasn't mentioned in the license agreement...now it is)

2 issues remain: 
a)Calling it Open Source or not...
b)clarification for SOCs/MSSPs. 

for now I put a clarification for SOCs/MSSPs in the above link. (is it good enough in your view? If not what is missing?)
for calling it "open source"....few clarification if I may..
Noun vs Verb
Open sourcing vs Open Source.
in your view do you object to both?
I understand the issue with the "Honor based system" of using the phrase "Open Source" only when using OSI licenses etc. because noone owns the TM on the phrase "Open Source" therefore can't force anyone to not use it.  Of course we want to make sure we continue to provide clarity required when communicating with the community hence my questions. We only want to make "positive contribution" :)

**Comment by nealmcb** [comment]
Thanks for taking care of the underlying license.

Re the SOC/MSSP clause: how do you legally define those sorts of organizations? What is there to prevent an arbitrary organization from incorporating those capabilities just in order to compete with you? What is there to reassure the legal team of one of them that they qualify, and you won't change your mind after they have committed to an OpenEDR-based approach?

Finally, back to "open source": the consensus use of the term over decades, regardless of the part of speech, has referred to the same set of freedoms that the OSI has codified. It isn't just bad manners to misuse the term. It misleads your users.

Just use an accurate term like "Source Available" instead.

I also ran across this cautionary tale from the very folks who drove the Commons Clause:

[Why Open Source Licenses with a Commons Clause May Become Less Common \| Finnegan](https://www.finnegan.com/en/insights/articles/why-open-source-licenses-with-a-commons-clause-may-become-less-common.html)

> Ultimately, Redis Labs decided to move away from the clause, identifying the following considerations: the term “Apache2 modified by Commons Clause” caused confusion with some users, who thought they were only bound by the terms of the Apache 2.0 license; the term “substantial” in the Common Clause was unclear; and the Commons Clause restriction regarding “support” appeared to work against Redis Labs’ intention of growing the ecosystem around Redis modules.

So that reinforces the other concerns I've shared above. The Commons Clause seems like a bad choice in the first place.

**Comment by ComodoMelih** [comment]
> Re the SOC/MSSP clause: how do you legally define those sorts of organizations? What is there to prevent an arbitrary organization from incorporating those capabilities just in order to compete with you? What is there to reassure the legal team of one of them that they qualify, and you won't change your mind after they have committed to an OpenEDR-based approach?

SOC/MSSP is a fairly well defined term and what they do is fairly clear. As long as you are a SOC/MSSP you can happily use OpenEDR to offer your SOC/MSSP services. If our competitor wants to offer SOC/MSSP services then they use OpenEDR within the same capacity. Its just they can't take the code and sell it as is and make money. We encourage SOCs and MSSPs to build a service around OpenEDR. That's the whole point of OpenEDR. 

As to "open source": we are aligned, I agree with you. I am aware of the term "Source Available" etc, we are thinking about how to refer to it.
Read that link already :) thanks for sharing.
Reddis have created their own version after removing Commons Clause. We are thinking on the same lines perhaps....We are working on it.

Here is a dilemma with Open Source as per OSI model: The way OSI has set it causes the following problem and would love to hear your view on it:
Company A : Great technical company
Company B: Great Marketing company

Company A creates an open source product, makes it open source. 
Company B takes that open source using its marketing muscle to generate a business/market for themselves.
This way innovator Company A loses out. So it seems that OSI model is good for predator marketing companies and not protecting innovators. 

How would Company A protect itself while Open sourcing?

**Comment by nealmcb** [comment]
From a community perspective, the goal is to have robust, widely supported software. The community benefits when multiple organizations successfully offer, sell, support and collaborate around a body of open source software.  Linux is a great example of that. Some companies are great at marketing, some at developing, some at adding nice interfaces, etc. They all contribute to the community by leveraging their own expertise. They compete vigorously, and many make lots of money. Linux would not have revolutionized the industry and created so much value if, like Unix, it had not been fully open source.

You've seen how open source itself as a development and community-building strategy has revolutionized the entire software industry. So have your competitors who offer truly open source EDR frameworks.

As in all walks of life, developing and executing a business plan in the software and support world is challenging. I appreciate you evolving your plan to involve more openness. But I'm not a business consultant, or expert in the EDR space, so I can't give you good advice on how to best do so.

I can simply encourage you to explore the benefits of the kind of fully engaged community you would get by adopting a fully open source route.  And if you don't choose that, I can help you describe your plan in ways that are accurate and let your collaborators quickly understand what they can and can't do with OpenEDR.

**Comment by ComodoMelih** [comment]
Thanks for the answer Neal. 
I do appreciate the value of a good ecosystem where everyone contributes from marketing companies to R&D teams. My question wasn't whether they had value or not, of course they all bring value. 
Also I am not arguing the value of "Open Source" vs "Closed Source". Of course exactly same two things, if you open source it will gain more traction. 
The pov/discussion was: How do you incentivize a business to open source using the OSI model? I want businesses to open source the code they write.
From my perspective, If I was to describe the model of OSI, I would say it lacks the incentive framework for businesses. You are encouraging a route of fully open sourcing but without offering a solution of how the company will pay its developers salaries. The OSI method will kill innovation coming from businesses and that is why Innovation is still closed source!  
The examples you give, Linux is exactly the proof! Where Linus as an individual wrote and released an OS (while in University in Helsinki. Although I don't know how it works, but I am assuming Linus was funded either by his government or the university to study and this time he was spending in writing the Linux code. Biggest example you give, Linux had its "funding source" coming from a "non for profit" organization eg: government or university) and now everyone else is making money from it. The brain created Linux wasn't able to benefit in a proportional scale of its scale. Therefore one could put forward an argument saying OSI model blunts innovation because it does not incentivize companies to open source using the OSI model.

**Comment by nealmcb** [comment]
I recommend more in-depth study of the history and economics of open source and Linux. To the surprise of many, nearly all of present-day LInux has come from for-profit companies, who introduce plenty of innovation and create value based on their investment in it.  Huge, innovative, successful companies like Red Hat, Intel, Suse, Samsung, or Oracle. Red Hat turned their Linux expertise into a company valued by IBM at **34 Billion dollars**, and last I heard, all of their products (including many applications and frameworks not in the kernel) have always been open source.

* [Who writes Linux? Corporations, more than ever \| InfoWorld](https://www.infoworld.com/article/2610207/who-writes-linux--corporations--more-than-ever.html)
* [Open source, the Red Hat way](https://www.redhat.com/en/open-source/red-hat-way)

And I think a very strong case can be made for the net benefits of open source to innovation, by opening the playing field up to everyone from kids at home to big corporations. See e.g. 

*  [11 Ways That Linux Contributes to Tech Innovation \- The Linux Foundation](https://www.linuxfoundation.org/blog/2015/03/11-ways-that-linux-contributes-to-tech-innovation/)
* [Open Source: A Platform for Innovation \| WIRED](https://www.wired.com/insights/2013/11/open-source-a-platform-for-innovation/)

Hugely successful projects like Raspberry Pi and RISC-V, which bring open source principles to hardware, show how much wider the idea can spread.

Again, it may or may not be a fit for your company. But learn and dream some more about the possibilities, and whatever you do, please give your customers the straight story.

**Comment by ComodoMelih** [comment]
One important thing you are missing is that none of "present-day Linux"  could have happened if it wasn't created by Linus in the first place! (which was funded by a "non for profit" ).
Again, I am not arguing the "after the fact" affects of open source...once something has been "open sourced". My point is businesses have no incentive to open source. 
Its the "start" ...."seed"...."first" is the issue...

Note: Raspberry Pi: business model was to sell hardware, thats how the people made money and I believe they were a "non for profit charity" (https://en.wikipedia.org/wiki/Raspberry_Pi )

Two examples of success was "created" (started/initiated) by a "non-for profit" organisations! Which is my point exactly. The OSI model has to change to incentivize businesses/corporations to want to "open source".

**Comment by nealmcb** [comment]
Look,  @ComodoMelih, you simply can't tell OSI what to do, or redefine the common understanding of "open source", for all the reasons I've articulated above.

You haven't fixed this bug, so please reopen it.

You can, as a separate effort, try to come up with some sort of "source available" license that you think works better for some kinds of businesses. It just isn't part of this particular issue.

I recommend working with @BrucePerens, a co-founder of OSI, and an expert in both business and licenses.
Your goals bring to mind the [Business Source License](https://perens.com/2017/02/14/bsl-1-1/) that Perens consulted on. The BSL is a non-Open-Source license which transitions into an Open Source license after a delay. 

But again, this bug remains, as Perens himself also notes: [they need to stop confusing it with open source!](https://perens.com/2018/08/31/redis-the-commons-clause-and-making-open-source-licenses-not-open-source/)
Please just fix your documentation claiming "open source" status, and get on with defining how you do want to work with potential collaborators.

**Comment by ComodoMelih** [comment]
@nealmcb
Quick summary: 
-You came here asking us to stop the term "Open Source" (although there is no legal right by OSI on this as they don't own the trademark)
-Its an honor system.
-We agreed to work with the honor system (and legal guys have prepared new agreement should be up next week hopefully) although its bad for innovation.
-Because you seem to be someone who cares about "Open Source' in general, so do I,  I wanted to share my views from a "Start-ups" perspective (you admitted not being a business guy), so exchanging ideas and perspective I thought would be healthy for both of us. We both want "innovation" to be "open sourced" was my assumption.  
-I wanted you to understand OSI is killing innovation! Was I able to explain it? I genuinely want to engage and exchange ideas and perspectives in an open and frank manner. 
-You gave examples of innovation that was funded by "non-profits", which was my point exactly! Innovation done by "for profits"  eg: businesses/start ups in the main, can't use the OSI model due to its model in my view.
All I wanted was for you to acknowledge the problem that OSI model has for a "for profit" model by giving you examples. 
Why?
-Because you care enough to come to companies like us and tell us not to use "open source", I thought maybe you care enough to understand the issues I raise and take it back to OSI. 
-Because I want you to understand by making anyone or any company change the wording from "open source" to something else (as you are asking us to do)  you are effectively diminishing the value of what they are doing. So effectively your actions @nealmcb  are anti-innovation! (using strong words only to get the message accross, I truly appreciate the work you are doing Neal!)
Do you acknowledge the anti innovation nature of OSI when it comes to "for profit" innovation Neal?  
If not, would you please answer the questions poised in my thought experiment so that I understand your thoughts.

**Comment by BrucePerens** [comment]
Hi @ComodoMelih,

I am co-founder of the Open Source Movement in software, and the main author of the Open Source Definition.

First, I would like to correct a fallacy in your summary:

> -I wanted you to understand OSI is killing innovation! 
OSI and other Open Source evangelists and promotors are _not killing innovation, and can not kill innovation_ because **nobody is compelled to call their software "Open Source".** It's just silly to assert otherwise. Please stop.

The "Commons Clause" is not compatible with the Open Source Definition, for the reasons already stated to you.

What it appears you are trying to do is make use of the "Open Source" imprimatur without complying with the definition. Regardless of the trademark status, this really looks to me like deceptive marketing of your software, please stop it until you can apply an unmodified license that is accepted as compliant with the Open Source Definition.

**Comment by ComodoMelih** [comment]
Hi @BrucePerens 
Firstly, thank you for taking the time to post here. Really appreciate it.
Secondly, as mentioned, we are in agreement with the "honor based system" OSI uses to get others not to call it "Open Source". Our team has been working on the agreements etc and hope to get it all done this coming week.
-Why OSI is killing innovation? (I apologize for using such strong terminology, its only meant to make a point nothing else, maybe I should have said, OSI is not contributing to  innovation by not incentivizing for profit organizations to open source their software?)
Let me explain it from my perspective: 
Because OSI model is not incentivizing "for profit" organizations to open their code. If for profit organizations were opening their code it would lead to a much higher degree of innovation in my view. Yes, I realize the complexities this will introduce in terms of licensing, but I am a firm believer that those complexities are worth the effort. 

@BrucePerens  would love to hear your views/thoughts on the following use case though please. I come from a thought process of "Nothing is perfect, therefore everything can be improved", including OSI licensing. Therefore thought provoking discussions/brain storming sessions will only help us all improve. Hope you will see this discussion in that light.

**Usecase:**
You have a great idea.
You need a team to write the code to bring this idea to reality (its a software/cloud product).
Its such an idea that once you put it out there people will start using it. (condition is: make public and you will have income coming in.)
It will take 2 years worth of exposure on your website to generate enough income to have a break even company where you can now afford to pay all your employees salaries.
You mortgaged your home and raised a loan from a bank and few friends and family gave you some money to start the business.
Question: Would you open source all that code at day one based on the OSI model?
- If the answer is YES: How could you make sure that others won't take all your creation and hard work and invest more marketing dollar than you invested to get more exposure and establish themselves, therefore you end up not reaching that break even and losing all your investment?
- If the answer is NO: What should OSI change for you to be comfortable releasing it at day one.

**Comment by BrucePerens** [comment]
We're going to confuse everyone if we call this the [OSI Model](https://en.wikipedia.org/wiki/OSI_model).

I am working on creating better incentives for revenue-seeking organizations to share their software in a way that sometimes produces revenue. The early draft license, which has not been blessed by a lawyer, is at https://perens.com/wp-content/uploads/sites/4/2020/10/PO_License_Draft.pdf and I discuss my rationale in the video at https://www.youtube.com/watch?v=vTsc1m78BUk

Open Source has already been incredibly successful, and through OSS Capital I hold enough interest in firms marketing Open Source products that there must be something viable there. However, invariably these are mixed-model firms. They find some way to mix Open Source and commercial licensing.

The most important thing about Open Source is that there is a well-defined set of rights. If someone calls it Open Source, you pretty much know what you are getting. This is _not_ the case with Creative Commons, where the right to read seems to be the only common element among the licenses.

It seems to me that there is value in preserving the coupling between the Open Source name and that well-defined set of rights, but it is not necessary for us to be constrained by it. Indeed, if we count Richard Stallman's September 27, 1983 announcement as the starting date, that's 37 years and two months. In that much time we should have learned something about it, and might come up with a successor. But in my thinking about any successor, it has seemed important to preserve "Open Source" in parallel to it, and not change the promise that has worked for so many people for all of that time.

I don't have a really well organized and legally sound tent with a motivating trademark for you to get under today. Business Source would not be a bad one. Monty wants you to use the same license they made, if you call it that.

**Comment by nealmcb** [comment]
Thanks, @ComodoMelih, for re-opening the issue again. I'm delighted to hear that you have updated language on the way and hope it resolves this particular issue. Please update us when you have a specific fix, before closing it again. A convenient way to do that is via a pull request.

Bruce is right that the stability of the OSD is hugely valuable, and asking them to change is quite out-of-place. I also agree with Bruce that it is silly to suggest that OSI is somehow harming innovation, and I provided plenty of reason to support the opposite conclusion.  Please note that I have not claimed that open source is best for all situations, and in that sense I'm not contradicting your claim that for your situation an OSI license might not fit. A proper analysis of that is something that as I've said, I'm not an expert in. But you certainly shouldn't expect me to agree with all your conclusions.

I hope you can help Bruce and others frame useful licenses that do in fact improve openness and innovation for additional business contexts.

**Comment by ComodoMelih** [comment]
> Bruce is right that the stability of the OSD is hugely valuable, and asking them to change is quite out-of-place. I also agree with Bruce that it is silly to suggest that OSI is somehow harming innovation, and I provided plenty of reason to support the opposite conclusion.

Let me give a coding analogy:
lets say OSI is software...and I am a user....I have a specific configuration that software doesn't work for...its a "BUG"....I report the bug to the "author".....calling the user "silly" for reporting the bug is an unfortunate choice of words. No user is silly for reporting bugs! Especially when the Author has already acknowledged the bug and is preparing a fix for it! 

> I am working on creating better incentives for revenue-seeking organizations to share their software in a way that sometimes produces revenue.

Here is a use case that noone is disputing: Once a code is open sourced: Innovation follows as a result.
I am NOT talking about this use case! (like I mentioned few times in my previous posts and you seem to conflate these two distinct use cases I am afraid :( ) 
The BUG being reported is: OSI does not incentivize businesses to open source their code.

Imagine this: If there were no obstacles for businesses to release their code as open source, then there would be more open source code and more innovation as a result.
Businesses are NOT releasing due to issues (BUG) I raised, therefore no innovation being built on top of these (unopened source code) as a result..."Lost Opportunity".....

**Comment by ComodoMelih** [comment]
> I am working on creating better incentives for revenue-seeking organizations to share their software in a way that sometimes produces revenue. The early draft license, which has not been blessed by a lawyer, is at https://perens.com/wp-content/uploads/sites/4/2020/10/PO_License_Draft.pdf and I discuss my rationale in the video at https://www.youtube.com/watch?v=vTsc1m78BUk

@BrucePerens  I very much enjoyed watching/reading the presentation!

**Begging loggers to plant and protect the trees!** 

Pretty much sums up the issue! Love the analogy btw :) 

I can see clearly from the presentation that you are very aware of the issues I am raising. 

Goal: Pay Developers!  <------Exactly my point! 

Question:
On this slide --> Goal: Serve the General Population
Is my understanding correct?
You are referring to developers who are writing open source code?
Because there are many Closed source developers that serve the general population, they just don’t have the incentive to release it as open source.
If that incentive was provided to turn closed source developer who serve general population into open source developer, then you would be achieving your goal.
Instead of getting developers who “scratch their own itch” and open source but don’t serve the general population
Get the developers who “serve the general population” but don’t open source to open source their code.

I believe OSI/Open Source is going through exactly same growth issues as any other product. 
https://en.wikipedia.org/wiki/Product_line_extension
It started with a single product, but now needs to create different "flavors" to satisfy different audiences.

we should definitely continue the discussion Bruce! OSI needs the line extension!

Ps: 48:30 : Typo “Carged” should be “Charged”

**Comment by BrucePerens** [comment]
Since post-Open-Source is peripheral to the nomenclature issue, let's take it off of the bug ticket system. Please resolve the nomenclature issue as requested.

In talking about serving the general population, the distinction I wanted to make is that a teenager using our software while going to college, and Amazon.com should be treated differently. We don't expect payment from the teenager, but should have a right to expect it from Amazon.com . Serving the developer community is implicit in the whole presentation.

It would probably be best to continue this via email to bruce@perens.com, however you are also welcome to call me, US-Pacific time, at 510-4PERENS. It takes a minute to get through the call screener the first time, then it will remember your number.

**Comment by jt0dd** [comment]
I just learned about the Commons Clause because of this issue and that is pretty interesting, basically a protection for the creators so that people can't leverage their code they made (for free, so it could benefit people.... for free) to charge people money to essentially use some form of it. I understand SaaS and the added value a company can offer with what they do with that code, but I do like the idea of the vendor being required to do far more than just re-package it, being required to offer to contribute any improvements they make to the project, etc etc.

**Comment by Kirutian** [comment]
Nextcloud is open source and is killing it in terms of innovation. Wordpress is another example. There are other examples as well. I'm a firm believer if you create an open source product that is useful and addresses a need, not only will it be adopted, you will get people to contribute to the project and you can demonstrably profit from it.
